Tests/IntegrationTests/PropagationTests.hpp

```diff
@@ -0,0 +1,477 @@
+
+  Vector3D pos = Vector3D::Zero();
+  double time = 0.0;
+  Vector3D mom = absMom * makeDirectionUnitFromPhiTheta(phi, theta);
+  NeutralCurvilinearTrackParameters params(std::nullopt, pos, mom, time);
+
+  // ensure initial parameters are valid
+  CHECK_CLOSE_ABS(params.position(), pos, 0.125_um);
+  CHECK_CLOSE_ABS(params.time(), time, 1_ps);
+  CHECK_CLOSE_ABS(params.momentum(), mom, 0.125_eV);
+  // charge should be identical not just similar
+  BOOST_CHECK_EQUAL(params.charge(), 0);
+
+  return params;
+}
+
+// helpers to compare track parameters
+
+/// Check that two parameters object are consistent within the tolerances.
+///
+/// \warning Does not check that they are defined on the same surface.
+template <typename charge_t>
+inline void checkParametersConsistency(
+    const Acts::SingleTrackParameters<charge_t>& cmp,
+    const Acts::SingleTrackParameters<charge_t>& ref, double epsPos,
+    double epsDir, double epsMom) {
+  using namespace Acts;
+
+  // check stored parameters
+  CHECK_CLOSE_ABS(cmp.template get<eBoundLoc0>(),
+                  ref.template get<eBoundLoc0>(), epsPos);
+  CHECK_CLOSE_ABS(cmp.template get<eBoundLoc1>(),
+                  ref.template get<eBoundLoc1>(), epsPos);
+  CHECK_CLOSE_ABS(cmp.template get<eBoundTime>(),
+                  ref.template get<eBoundTime>(), epsPos);
+  // check phi equivalence with circularity
+  CHECK_SMALL(detail::radian_sym(cmp.template get<eBoundPhi>() -
+                                 ref.template get<eBoundPhi>()),
+              epsDir);
+  CHECK_CLOSE_ABS(cmp.template get<eBoundTheta>(),
+                  ref.template get<eBoundTheta>(), epsDir);
+  CHECK_CLOSE_ABS(cmp.template get<eBoundQOverP>(),
+                  ref.template get<eBoundQOverP>(), epsMom);
+  // check derived parameters
+  CHECK_CLOSE_ABS(cmp.position(), ref.position(), epsPos);
+  CHECK_CLOSE_ABS(cmp.time(), ref.time(), epsPos);
+  CHECK_CLOSE_ABS(cmp.momentum(), ref.momentum(), epsMom);
+  // charge should be identical not just similar
+  BOOST_CHECK_EQUAL(cmp.charge(), ref.charge());
+}
+
+/// Check that two parameters covariances are consistent within the tolerances.
+///
+/// \warning Does not check that the parameters value itself are consistent.
+template <typename charge_t>
+inline void checkCovarianceConsistency(
+    const Acts::SingleTrackParameters<charge_t>& cmp,
+    const Acts::SingleTrackParameters<charge_t>& ref,
+    double relativeTolerance) {
+  BOOST_CHECK(
+      not(cmp.covariance().has_value() xor ref.covariance().has_value()));
+  if (cmp.covariance().has_value() and ref.covariance().has_value()) {
+    CHECK_CLOSE_COVARIANCE(cmp.covariance().value(), ref.covariance().value(),
+                           relativeTolerance);
+  }
+}
+
+// helpers to construct target surfaces from track states
+
+/// Construct the transformation from the curvilinear to the global coordinates.
+template <typename charge_t>
+inline std::shared_ptr<Acts::Transform3D> makeCurvilinearTransform(
```

Comment:
Because this can take arbitrary bound parameters as input, but I want the constructed surface to always be perpendicular to the track direction. Using the curvilinear transform is the simplest way to get this because it automatically defines the two local coordinates.

Reply:
Why would you use a perpendicular transformation for a given surface?

Reply:
I want a surface for which I know that the track can intersect it. The easiest way to ensure that, is to use a surface that is perpendicular to the track direction at the point of intersection. because there distance to the surface changes maximally with the propagation length. That however defines only a normal direction and I need two additional unit vectors to construct the local system.

Reply:
Ok, got it. The name is just quite misleading if you read it with a certain assumption of what it should do.